File: frontend/src/types/queue.ts
export interface CabinetOut {
  id: number
  code: string
  name: string
  modality: string
  is_active: boolean
}

export interface PatientOut {
  id: string
  full_name: string
  policy_number: string
  birth_date: string | null
  phone: string | null
  created_at: string
}

export interface TicketOut {
  id: number
  ticket_number: string
  status: string
  cabinet_id: number
  patient_id: string
  order_id: string | null
  study_uid: string | null
  called_at: string | null
  completed_at: string | null
  created_at: string
}

export interface TicketDetail extends TicketOut {
  patient: PatientOut
  cabinet: CabinetOut
  priority?: 'normal' | 'urgent' | 'stat' | null
  sourceTicketId?: string
  service_type_name?: string
  modality?: string
}

export interface TicketEventOut {
  id: number
  ticket_id: number
  event_type: string
  from_status: string | null
  to_status: string | null
  created_at: string
}

export interface TicketCreateRequest {
  full_name: string
  policy_number: string
  cabinet_code: string
  phone?: string | null
  birth_date?: string | null
  priority?: 'normal' | 'urgent' | 'stat'
}

export interface NextCallRequest {
  cabinet_code: string
}

export interface RisRoomOut {
  id: number
  name: string
  is_active: boolean
  modality: string
}

export interface RisServiceTypeOut {
  id: number
  name: string
  modality_code: string
  average_duration_minutes: number
}

export interface RisTicketOut {
  id: string
  ticket_number: string
  status: string
  priority: string
  modality: string
  cabinet_id: number
  cabinet_name: string
  full_name: string
  policy_number: string
  service_type_name?: string
  order_id?: string | null
  study_uid?: string | null
  created_at: string
  called_at: string | null
  completed_at: string | null
}

File: backend/db/schemas/queue.py
"""DTO электронной очереди."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CabinetOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    modality: str
    is_active: bool


class PatientCreate(BaseModel):
    full_name: str = Field(min_length=1, max_length=255)
    policy_number: str = Field(min_length=1, max_length=64)
    birth_date: datetime | None = None
    phone: str | None = Field(default=None, max_length=32)
    notes: str | None = None


class PatientOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    full_name: str
    policy_number: str
    birth_date: datetime | None
    phone: str | None
    created_at: datetime


class TicketCreateRequest(BaseModel):
    """Запрос на регистрацию пациента в очереди."""

    full_name: str = Field(min_length=1, max_length=255)
    policy_number: str = Field(min_length=1, max_length=64)
    cabinet_code: str = Field(default="101", description="Код кабинета (101, 102, …)")
    phone: str | None = None
    birth_date: datetime | None = None
    priority: str = Field(default="normal", pattern=r"^(normal|urgent|stat)$",
                          description="Приоритет: normal / urgent / stat")


class TicketOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    ticket_number: str
    status: str
    cabinet_id: int
    patient_id: uuid.UUID
    order_id: str | None
    study_uid: str | None
    called_at: datetime | None
    completed_at: datetime | None
    created_at: datetime


class TicketDetail(TicketOut):
    """Талон + сведения о пациенте и кабинете (для UI)."""

    patient: PatientOut
    cabinet: CabinetOut
    priority: str | None = None  # из связанного Order (scheduled, urgent, stat)


class TicketEventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    ticket_id: int
    event_type: str
    from_status: str | None
    to_status: str | None
    created_at: datetime


class NextCallRequest(BaseModel):
    cabinet_code: str = Field(default="101")


class CompleteRequest(BaseModel):
    ticket_number: str
    protocol: str | None = Field(default=None, max_length=8192)
